webapp/channels/src/components/admin_console/session_length_settings.tsx

```diff
@@ -4,16 +4,34 @@
 import React from 'react';
 import {FormattedMessage} from 'react-intl';
 
+import {AdminConfig, ServiceSettings} from '@mattermost/types/config';
+
 import FormattedMarkdownMessage from 'components/formatted_markdown_message';
 import * as Utils from 'utils/utils';
 
-import AdminSettings from './admin_settings';
+import AdminSettings, {BaseState, BaseProps} from './admin_settings';
 import BooleanSetting from './boolean_setting';
-import SettingsGroup from './settings_group.jsx';
+import SettingsGroup from './settings_group';
 import TextSetting from './text_setting';
 
-export default class SessionLengthSettings extends AdminSettings {
-    getConfigFromState = (config) => {
+interface SessionLengthSettingsState extends BaseState {
+    extendSessionLengthWithActivity: ServiceSettings['ExtendSessionLengthWithActivity'];
+    sessionLengthWebInHours: ServiceSettings['SessionLengthWebInHours'];
+    sessionLengthMobileInHours: ServiceSettings['SessionLengthMobileInHours'];
+    sessionLengthSSOInHours: ServiceSettings['SessionLengthSSOInHours'];
+    sessionCacheInMinutes: ServiceSettings['SessionCacheInMinutes'];
+    sessionIdleTimeoutInMinutes: ServiceSettings['SessionIdleTimeoutInMinutes'];
+    sessionIdleTimeoutMobileInMinutes: number;
+}
+
+interface SessionLengthSettingsProps extends BaseProps {
```

Comment:
instead of this you can use already existing `ClientLicense`

Reply:
@harshal2030 this is not addressed, can you please look